### PR TITLE
Set TNT source in blast mining

### PR DESCRIPTION
**Pull request by GriffinCodes**
Using the source in my own plugin; I know the metadata key exists, but I'd rather just use the provided bukkit method

### CODE REVIEW

**Comment by lexikiq** [comment]
bump! would appreciate this feature as well. would be nice for logging systems, like grief detection or otherwise

**Review by lexikiq — APPROVED** [review]

File: src/main/java/com/gmail/nossr50/skills/mining/MiningManager.java [hunk 1]
```diff
@@ -119,6 +119,7 @@ public void remoteDetonation() {
 
         tnt.setMetadata(mcMMO.tntMetadataKey, mmoPlayer.getPlayerMetadata());
         tnt.setFuseTicks(0);
+        tnt.setSource(player);
```
Comment (TheBusyBiscuit) [inline src/main/java/com/gmail/nossr50/skills/mining/MiningManager.java:122]:
```suggestion
        if (mcMMO.getCompatibilityManager().getMinecraftGameVersion().isAtLeast(1, 16, 4)) {
            tnt.setSource(player);
        }
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
There is a huge issue with this though.

The method `TNTPrimed#setSource()` was added on Oct 25th 2020, source:
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/4abf9e96df578e4c842c865e50d0933ce0616938#src/main/java/org/bukkit/entity/TNTPrimed.java

In other words, this method was added in 1.16.3 (-SNAPSHOT, 1.16.4 was the first proper release to have included this) and won't be available in prior Minecraft versions.
That's why I am suggesting a version-gated check here, to make sure there won't be any `NoSuchMethodException`s spit out to servers on earlier versions.

**Review by TheBusyBiscuit — APPROVED** [review]

**Comment by nossr50** [comment]
Thank you

**Pull request merged**